htlcswitch/link.go

```diff
@@ -2335,6 +2335,15 @@ func (l *channelLink) canSendHtlc(policy ForwardingPolicy,
 		return &lnwire.FailExpiryTooFar{}
 	}
 
+	// Check to see if there is enough balance in this channel.
```

Comment:
Commit message leaves some to be desired, a bug is actually being addressed here, but the commit message instead conveys some generic "clean up". Which cases were previously incorrectly handled as a result of the existing logic?

Reply:
Continuing, if a bug is indeed being fixed here, then the fix should be accompanied by a test. IIUC, the prior logic would cause us to send `UnknownNextPeer` in the scenario that there're multiple links to pick from with one of them having insufficient bandwidth and the other triggering a forwarding policy fail error (diff time lock that the other channel for example).

Reply:
Commit message fixed, tests added.